acceptance_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connector
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/conduitio/conduit-connector-google-cloudstorage/source"
+	"github.com/google/uuid"
+	"go.uber.org/goleak"
+
+	"github.com/conduitio/conduit-connector-google-cloudstorage/config"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type GCSAcceptanceTestDriver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+	GCSClient *storage.Client
+}
+
+func (d GCSAcceptanceTestDriver) WriteToSource(t *testing.T, records []sdk.Record) []sdk.Record {
+	ctx := context.Background()
+	testBucket := d.SourceConfig(t)[config.ConfigKeyGCSBucket]
+	for _, record := range records {
+		wc := d.GCSClient.Bucket(testBucket).Object(string(record.Key.Bytes())).NewWriter(ctx)
+		fmt.Fprint(wc, string(record.Payload.Bytes()))
```

Comment:
Not a blocker, asking out of curiosity: isn't it possible to write bytes directly, without converting to a string first.

Reply:
Yes we can, *storage.Writer is an implementation of io.Writer only, So it contains a Write method which accept bytes.

Reply:
Applied its Done